ext/civi_case/civi_case.php

```diff
@@ -1,3 +1,18 @@
 <?php
 
 require_once 'civi_case.civix.php';
+use CRM_Case_ExtensionUtil as E;
+
+/**
+ * Implements hook_civicrm_managed().
+ */
+function civi_case_civicrm_managed(&$entities, $modules) {
+  // Historically these were considered part of "civicrm" not "civi_case", so accept either
```

Comment:
Digging into the definitions from `CRM_Case_ManagedEntities`...

* In `createManagedCaseTypes()`, the ownership is [passed-through](https://github.com/colemanw/civicrm-core/blob/3a5553fa5980cd2269ab6da8155ca0de26340628/CRM/Case/ManagedEntities.php#L36) from whatever module defined the case-type.
* In `createManagedRelationshipType()` and `createManagedActivityTypes()`, it ascribes ownership to module `civicrm`.

I guess the main thing is the ownership doesn't change (i.e. we won't delete+reinsert). This conditional here is an optimization saying: "We don't need to lookup some mgds."

But you probably need to delegate to `createManagedCaseTypes()` unconditionally, right? (Since you can't predict which modules it speaks on behalf of...)

Reply:
@totten yes I see the delegation happening with `hook_civicrm_caseTypes()` to other modules.
Ok I've removed the optimization from the hook.

Aside... I wonder if it's time to noisily deprecate `hook_civicrm_caseTypes` in favor of `hook_civicrm_managed`